src/test/test-execute.c

```diff
@@ -218,6 +219,41 @@ static void start_parent_slices(Unit *unit) {
         }
 }
 
+static bool can_use_unprivileged_userns(void) {
```

Comment:
I now realize that above you *were* talking about the `kernel.unprivileged_userns_clone` sysctl. This is actually a [Debian patch](https://salsa.debian.org/kernel-team/linux/-/blob/master/debian/patches/debian/add-sysctl-to-disallow-unprivileged-CLONE_NEWUSER-by-default.patch?ref_type=heads), but I thought it was mainline partially because it's mentioned all over [`systemd.exec`](https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html).

So on a system with that patch, this function really does check if you can call `unshare(CLONE_NEWUSER)` without `CAP_SYS_ADMIN`.